resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/ServerFormUrlEncodedProvider.java

```diff
@@ -1,52 +1,57 @@
-package org.jboss.resteasy.plugins.providers;
-
-import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
-import org.jboss.resteasy.spi.HttpRequest;
-import org.jboss.resteasy.spi.util.FindAnnotation;
-
-import javax.ws.rs.ConstrainedTo;
-import javax.ws.rs.Consumes;
-import javax.ws.rs.Encoded;
-import javax.ws.rs.Produces;
-import javax.ws.rs.RuntimeType;
-import javax.ws.rs.core.Context;
-import javax.ws.rs.core.MediaType;
-import javax.ws.rs.core.MultivaluedMap;
-import java.io.IOException;
-import java.io.InputStream;
-import java.lang.annotation.Annotation;
-import java.lang.reflect.Type;
-
-/**
- * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
- * @version $Revision: 1 $
- */
-@Produces("application/x-www-form-urlencoded")
-@Consumes("application/x-www-form-urlencoded")
-@ConstrainedTo(RuntimeType.SERVER)
-public class ServerFormUrlEncodedProvider extends FormUrlEncodedProvider
-{
-   protected boolean useContainerParams;
-
-   public ServerFormUrlEncodedProvider(final boolean useContainerParams)
-   {
-      this.useContainerParams = useContainerParams;
-   }
-
-   @Context
-   HttpRequest request;
-
-
-   @SuppressWarnings("rawtypes")
-   @Override
-   public MultivaluedMap readFrom(Class<MultivaluedMap> type, Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, String> httpHeaders, InputStream entityStream) throws IOException
-   {
-      LogMessages.LOGGER.debugf("Provider : %s,  Method : readFrom", getClass().getName());
-      if (!useContainerParams) return super.readFrom(type, genericType, annotations, mediaType, httpHeaders, entityStream);
-
-      boolean encoded = FindAnnotation.findAnnotation(annotations, Encoded.class) != null;
-      if (encoded) return request.getFormParameters();
-      else return request.getDecodedFormParameters();
-
-   }
-}
+package org.jboss.resteasy.plugins.providers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import javax.ws.rs.ConstrainedTo;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Encoded;
+import javax.ws.rs.Produces;
+import javax.ws.rs.RuntimeType;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.jboss.resteasy.resteasy_jaxrs.i18n.LogMessages;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.jboss.resteasy.spi.util.FindAnnotation;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+@Produces("application/x-www-form-urlencoded")
+@Consumes("application/x-www-form-urlencoded")
+@ConstrainedTo(RuntimeType.SERVER)
+public class ServerFormUrlEncodedProvider extends FormUrlEncodedProvider
+{
+   protected boolean useContainerParams;
+
+   public ServerFormUrlEncodedProvider()
```

Comment:
@triceo Can you please share the details about this change? This doesn't seems to be related with Java's ServiceLoader SPI described in [RESTEASY-2109](https://issues.jboss.org/browse/RESTEASY-2109) ...

Reply:
@marekkopecky I went through all the SPI uses, looking for a situation where:

- Some constructor is provided,
- while no nullary constructor is provided.

Since this would cause the default nullary constructor not to be inserted by the JVM, `native-image` would have failed with the same issue after the earlier issue was fixed.

However, now that I'm double-checking my work, I see that the `ServerFormUrlEncodedProvider` is not actually made accessible through the SPI and this change is therefore unnecessary. Thanks for pointing this out.

I will remove this change when I'm updating the PR to fix the whitespace changes.